src/fastgltf.cpp

```diff
@@ -2181,8 +2181,51 @@ bool fg::GltfDataBuffer::copyBytes(const std::uint8_t* bytes, std::size_t byteCo
     return true;
 }
 
-#if __ANDROID__
-bool fg::GltfDataBuffer::loadFromFile_Android(const fs::path& path, std::uint64_t byteOffset) noexcept {
+bool fg::GltfDataBuffer::loadFromFile(const fs::path& path, std::uint64_t byteOffset) noexcept {
+    using namespace simdjson;
+    // Open the file and determine the size.
+    std::ifstream file(path, std::ios::binary);
+    if (!file.is_open() || file.bad())
+        return false;
+
+    filePath = path;
+
+    // Skip over as many characters as possible until EOF, then get how many characters were skipped.
+    file.ignore(std::numeric_limits<std::streamsize>::max());
+    auto length = file.gcount() - byteOffset;
+    if (length == 0 || file.bad())
+        return false;
+
+    file.clear();
+    file.seekg(static_cast<std::streamsize>(byteOffset), std::ifstream::beg);
+
+    dataSize = static_cast<std::uint64_t>(length) - byteOffset;
+    allocatedSize = dataSize + simdjson::SIMDJSON_PADDING;
+    buffer = decltype(buffer)(new std::uint8_t[allocatedSize]); // To mimic std::make_unique_for_overwrite (C++20)
+    if (!buffer)
+        return false;
+    bufferPointer = buffer.get();
+
+    // Copy the data and fill the padding region with zeros.
+    file.read(reinterpret_cast<char*>(bufferPointer), static_cast<std::streamsize>(dataSize));
+    std::memset(bufferPointer + dataSize, 0, allocatedSize - dataSize);
+    return true;
+}
+
+std::size_t fg::GltfDataBuffer::getBufferSize() const noexcept {
+    return dataSize;
+}
+#pragma endregion
+
+#pragma region AndroidGltfDataBuffer
+#if defined(__ANDROID__)
+fg::AndroidGltfDataBuffer::AndroidGltfDataBuffer(AAssetManager* asset_manager_in) noexcept : asset_manager{asset_manager_in} {}
+
+bool fg::AndroidGltfDataBuffer::loadFromAndroidAsset(const fs::path& path, std::uint64_t byteOffset) noexcept {
+    if (asset_manager == nullptr) {
+        return false;
+    }
```

Comment:
Shouldn't this be a exception? (Or assert, but I would dislike asserts because those should only be used for checks of internal state, not something the user could break)

Reply:
I wasn't sure if you were exception-friendly

Reply:
Right not, no. But I am thinking of changing that to some core parts of the library because simdjson by default usually also uses exceptions. I can imagine that using exceptions could save a few things here and there. I guess it's fine as is and I will change it myself later if I do switch to exceptions.